src/test/java/uk/gov/hmcts/reform/notificationservice/service/NotificationMessageHandlerTest.java

```diff
@@ -0,0 +1,120 @@
+    }
+
+    @Test
+    public void should_notify_for_successful_notification_message() {
+        NotificationMsg notificationMsg =
+            new NotificationMsg(
+                "Zipfile.zip",
+                "probate",
+                "A123",
+                "A1342411414214",
+                ErrorCode.ERR_AV_FAILED,
+                "error description Av not valid",
+                "processor"
+            );
+
+        ErrorNotificationRequest request =
+            new ErrorNotificationRequest(
+                "Zipfile.zip",
+                "A123",
+                "ERR_AV_FAILED",
+                "error description Av not valid"
+            );
+
+        when(errorNotificationRequestMapper.map(notificationMsg)).thenReturn(request);
+        when(errorNotificationClient.notify(request))
+            .thenReturn(new ErrorNotificationResponse("312313-22311231-21321"));
+
+        notificationMessageHandler.handleNotificationMessage(notificationMsg);
+
+        // then
+        verify(errorNotificationRequestMapper).map(notificationMsg);
+        verify(errorNotificationClient).notify(request);
+    }
+
+    @Test
+    public void should_rethrow_feign_exception_when_notification_call_fails() {
+        NotificationMsg notificationMsg =
+            new NotificationMsg(
+                "file.txt",
+                "sscs",
+                "123213",
+                "32313223",
+                ErrorCode.ERR_SERVICE_DISABLED,
+                "error description service disabled",
+                "processor"
+            );
+
+        ErrorNotificationRequest request =
+            new ErrorNotificationRequest(
+                "file.txt",
+                "123213",
+                "ERR_SERVICE_DISABLED",
+                "error description service disabled"
+            );
+        when(errorNotificationRequestMapper.map(notificationMsg)).thenReturn(request);
+
+        FeignException exception = mock(FeignException.class);
+        doThrow(exception).when(errorNotificationClient).notify(request);
+
+        // when
+        assertThatThrownBy(() -> notificationMessageHandler.handleNotificationMessage(notificationMsg))
+            .isSameAs(exception);
+    }
+
+    @Test
+    public void should_rethrow_invalid_message_exception_when_parsing_fails() {
+        NotificationMsg notificationMsg =
+            new NotificationMsg(
+                "file.txt",
+                "sscs",
+                "123213",
+                "32313223",
+                ErrorCode.ERR_SERVICE_DISABLED,
+                "error description service disabled",
+                "processor"
+            );
+
+        InvalidMessageException exception = new InvalidMessageException("Parsed Failed");
+        doThrow(exception).when(errorNotificationRequestMapper).map(notificationMsg);
+
+        // when
+        assertThatThrownBy(() -> notificationMessageHandler.handleNotificationMessage(notificationMsg))
+            .isSameAs(exception);
+    }
```

Comment:
should we add assertions that no other mock interactions are excepted?

Reply:
added